### ISSUE TITLE
Add notes everywhere, starting with surveys

### ISSUE THREAD

**Issue by jmduke** [buttondown/roadmap#1860]
## TL;DR

Buttondown has a simple concept of "notes": free-form private text that you can attach to subscribers. We should add this to a lot more stuff, starting with **surveys**, so that authors can keep track of their Buttondown data a little bit better.

## Background notes

- You should start by getting yourself familiar with how notes work for Subscribers. They're pretty basic: just a free-form text area on the `Subscriber` model called `notes`. Search for all the plumbing in the codebase that reference it, and try adding/removing notes for a test subscriber in your local devbox. 
- Same thing with surveys! Try creating a survey and just seeing what methods get called, where the data is, etc.

## Useful files

- [The subscriber model](https://github.com/buttondown-email/monorepo/blob/aa30a4c85d94c6960fd669b3bbb979241a159a8d/app/emails/models/subscriber/model.py#L4), as a useful reference for how notes work
- [The survey model](https://github.com/buttondown-email/monorepo/blob/aa30a4c85d94c6960fd669b3bbb979241a159a8d/app/emails/models/subscriber/model.py#L4), where we'll be adding a notes field
- [The API endpoints that expose surveys to users](https://github.com/buttondown-email/monorepo/blob/aa30a4c85d94c6960fd669b3bbb979241a159a8d/app/api/views/surveys.py)
- [The Vue component where you can modify Subscriber notes (it's messy, sorry!)](https://github.com/buttondown-email/monorepo/blob/aa30a4c85d94c6960fd669b3bbb979241a159a8d/app/assets/components/Screens/Subscribers/Detail/Component.vue)
- [The Vue component where you can modify Survey notes (less messy!)](https://github.com/buttondown-email/monorepo/blob/aa30a4c85d94c6960fd669b3bbb979241a159a8d/app/assets/components/Screens/Surveys/View/Container.vue)

## Suggested breakdown

- [ ] Add a field on the `Survey` model called `notes`. Create the relevant migration for it.
- [ ] Expose this new model via the API (`api.views.surveys`). Be sure to add tests to make sure it works!
- [ ] Expose notes in the frontend UI so authors can set notes. No explicit tests necessary, but definitely make sure to try it out to make sure it works!

## Follow-on models

- [ ] Tags
- [ ] Emails
- [ ] Webhooks
